src/actions/guides/frontend/html_forms.cr

```diff
@@ -0,0 +1,533 @@
+class Guides::Frontend::HtmlForms < GuideAction
+  guide_route "/frontend/html-forms"
+
+  def self.title
+    "HTML Forms"
+  end
+
+  def markdown : String
+    <<-MD
+    ### Working with Lucky's HTML builder
+
+    All HTML tags have an associated method based on the name of the tag, with the exception of a few.
+    (e.g. `<form></form>` - `form()`, `<button></button>` - `button()`, etc...)
+
+    These are the only HTML tags with an alias name method.
+
+    * `<select></select>` - `select_tag`
+    * `<p></p>` - `para`
+
+    > These are overridden as to not clash with built-in Crystal methods.
+
+    In addition to being able to specify any form element by calling that method, Lucky also
+    gives you access to some helper methods to make this easier. All of these methods will go
+    in your `content` method of your page, or in a component.
+
+    ## Form tag
+
+    The `form_for` method takes any route or [Action Class](#{Guides::HttpAndRouting::RoutingAndParams.path}), any html options, and a block
+    to encompass the whole form.
+
+    As well as defining the `<form>` tag for you, the `form_for` method will also include an optional
+    `csrf_hidden_input` by default. To disable this option, look in your `config/form_helpers.cr`.
+
+    ```html
+    <form method="post" action="/posts" class="inline-form">
+      <input type="hidden" name="_csrf" value="some_token" />
+    </form>
+    ```
+
+    becomes
+
+    ```crystal
+    form_for(Posts::Create, class: "inline-form") do
+    end
+    ```
+
+    The Action class already has a route defined that sets the HTTP METHOD. When passing
+    that class in, Lucky will know which method to set for you.
+
+    You can also pass in a route path if you need to pass in additional URL params.
+
+    ```crystal
+    form_for(Posts::Create.with(author_id: @current_user.id)) do
+    end
+    ```
+
+    ### Custom forms
+
+    If you need more control over how your form is displayed, you can always use the `form`
+    method directly, and pass any options you'd like.
+
+    ```crystal
+    form(method: "get", class: "custom-form", action: Search::Index.path) do
+    end
+    ```
+
+    ## Input fields
+
+    All of the input helper methods take an `Avram::Attribute`. These are created
+    from setting an `attribute` or `permit_columns` in your `Avram::Operation`.
+    Then calling that attribute on the Operation.
+    See the [Operations Guide](#{Guides::Database::ValidatingSaving.path}) for more info.
+
+    > For these examples, `op` will refer to an instance of an `Avram::Operation`.
```

Comment:
```suggestion
    > For these examples, `op` will refer to an instance of an `Avram::Operation` (such as `SaveUser`).
```

Thoughts on the example? Helpful or harmful? I'm fine either way but wanted to see what you think